src/tdi_json_parser/tdi_info_parser.cpp

```diff
@@ -538,6 +544,23 @@ std::unique_ptr<tdi::TableInfo> TdiInfoParser::parseTable(
               __LINE__,
               table_id);
   }
+
+  // update relevant maps
+  for (const auto &kv : table_info->table_key_map_) {
+    const auto key_field = kv.second.get();
+    table_info->name_key_map_[key_field->nameGet()] = key_field;
+  }
+
+  for (const auto &kv : table_info->table_action_map_) {
+    const auto action = kv.second.get();
+    table_info->name_action_map_[action->nameGet()] = action;
+  }
+
+  for (const auto &kv : table_info->table_data_map_) {
+    const auto data_field = kv.second.get();
+    table_info->name_data_map_[data_field->nameGet()] = data_field;
+  }
+
```

Comment:
Is it possible to do this inside the ctor itself ?

Reply:
It is possible. But it involves maintaining a local copy of these maps and passing them to the table_info constructor.
I think filling them outside the constructor is simpler.

Reply:
Moved the map update code to the constructor.